extensions/pkg/terraformer/types.go

```diff
@@ -96,7 +97,9 @@ const (
 type Terraformer interface {
 	UseV2(bool) Terraformer
 	SetLogLevel(string) Terraformer
+	// Deprecated: use SetEnvVars instead
```

Comment:
I would like to highlight the fact that It seems the behaviour was changed from a "hard" set to an "update/append".  The same behaviour is used for the replacement `SetEnvVars`. The main API loss here is the ability to remove keys by setting a new map. 

Disclaimer: I haven't checked all usages of the (newly) deprecated function.

Reply:
> behaviour was changed from a "hard" set to an "update/append" [...] vs out-right removal from the interface is better

Yes, that observation is correct. I mainly did it via `append`, so users can use both functions in parallel. Though, I don't know if there's even a use case for it. And I also only deprecated the old func, so that users of the library are not put under pressure to transition/migrate. In general, I like to allow such smooth transitions wherever possible and I don't see a problem with it here.

> The main API loss here is the ability to remove keys by setting a new map

Yes, that's also true. But I guess, there is no need for it.

Side-note: This comes with an `action` release note and when developers vendor such a new gardener release, they are responsible for checking if their code is still compatible (at best with a good suite of unit tests). So I don't see any problem with the changed behaviour, as the difference is quite subtle.

But if you see any problems with it, please bring them up :)

Reply:
My (somewhat theoretical) how-should-it-be-handled PoV here was that given the subtle change of behaviour in the deprecated function we could:
a) make the `envVar` translation but keep the hard-set for full backwards compatibility
b) outright delete the func to force an action during the revendoring. 

But in practice I haven't seen such a use in our repos, so no objections on my side. The `action` note (which admittedly I missed on my first read) does the job.